test/voting.spec.ts

```diff
@@ -0,0 +1,138 @@
+
+const web3 = new Web3(Web3.givenProvider);
+
+contract("VotingContract", async accounts => {
+    let managerInstance: ManagerContractInstance;
+    let categoryInstance: CategoryContractInstance;
+    const categoryName = "FizzBuzz";
+
+    before(async () => {
+        managerInstance = await ManagerContract.deployed();
+        expect(managerInstance).to.be.not.null.and.not.undefined;
+    });
+
+    context("In a public voting with 10 options", async () => {
+        let votingContract: VotingContractInstance;
+        const votingEndTime = Math.floor(Date.now() / 1000) + 25;
+        const resultsEndTime = votingEndTime + 25;
+
+        const userOneOption = 4;
+        const userTwoOption = 7;
+
+        before(async () => {
+            const createVotingTxResp = await managerInstance.createVotingWithNewCategory(
+                categoryName,
+                "Pick your favourite letter",
+                ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J"],
+                votingEndTime,
+                resultsEndTime,
+                false,
+                [],
+                { from: accounts[0] }
+            );
+            expect(createVotingTxResp.logs[0].event).equals("CategoryCreated");
+            expect(createVotingTxResp.logs[1].event).equals("VotingCreated");
+            categoryInstance = await CategoryContract.at(
+                createVotingTxResp.logs[0].args.categoryAddress
+            );
+            expect(categoryInstance).to.be.not.null.and.not.undefined;
+            const votingAddress: string =
+                createVotingTxResp.logs[1].args.votingAddress;
+            votingContract = await VotingContract.at(votingAddress);
+            expect(votingContract).to.be.not.null.and.not.undefined;
+        });
+
+        describe("#isPrivate", async () => {
+            it("returns false", async () => {
+                const isPrivate = await votingContract.isPrivate();
+                expect(isPrivate).to.equal(false);
+            });
+        });
+
+        describe("#numberOfOptions", async () => {
+            it("returns 10", async () => {
+                const numberOfOptions = await votingContract.numberOfOptions();
+                expect(numberOfOptions.toNumber()).to.equal(10);
+            });
+        });
+
+        describe("#vote", async () => {
+            const userOne = accounts[0];
+            const userTwo = accounts[1];
+            context("when user #1 votes for the first time", async () => {
+                it("does not throw", async () => {
+                    await votingContract.vote(userOneOption, { from: userOne });
+                });
+            })
+
+            context("when user #1 tries to vote second time for the same option", async () => {
+                it("reverts", async () => {
+                    await truffleAssert.fails(
+                        votingContract.vote(userOneOption, { from: userOne }),
```

Comment:
Is it possible to check if the error message from VC "You have already voted" is properly returned? If not that's k, if it is then add it if it does not require too much effort.

Reply:
It's not currently possible

Reply:
Actually, that's not true -- it **is** possible, and I'll add that :whale:

Reply:
Damn it, I was right the first time -- there is no message coming through truffle...